### ISSUE TITLE
Override fetch function for SQL-over-HTTP

### ISSUE THREAD

**Issue by TimoWilhelm** [neondatabase/serverless#34]
Currently the only way to modify the `fetch` behavior is through the `fetchOptions` param.

It might also make sense to allow overriding of the `fetch` function itself. Some SSR frameworks provide alternative fetch functions with additional features or better performance and it could make it easier to extend the fetch behaviour (for example using [fetch-retry](https://www.npmjs.com/package/fetch-retry)).

Anyway, the new new SQL-over-HTTP driver looks really promising!

**Comment by jawj** [comment]
This is a nice idea. I already had vaguely in mind that we should provide something equivalent to the `webSocketConstructor` option, but for the http case. I'll see to it.

**Comment by jawj** [comment]
This is released on 0.5.3 and works like this:

```typescript
neonConfig.fetchFunction = (url: string, options: any) => {
  console.log('custom fetch:', url, options);
  return fetch(url, options);
};
await sql`SELECT ${"customFetch"} AS str`;
```

Which logs:

```
custom fetch: https://ep-xyz.eu-central-1.aws.neon.tech/sql {
  method: 'POST',
  body: '{"query":"SELECT $1 AS str","params":["customFetch"]}',
  headers: {
    'Neon-Connection-String': 'postgres://user:pass@xyz.eu-central-1.aws.neon.tech/main',
    'Neon-Raw-Text-Output': 'true',
    'Neon-Array-Mode': 'true',
    'Neon-Pool-Opt-In': 'true'
  }
}
```
